challenges/largest-product-array/largest_product.py

```diff
@@ -0,0 +1,44 @@
+def largest_product(arr):
+    '''Returns the largest product of adjacent values in an m x n matrix, given m > 1, n > 1'''
+
+    # Check param type and values
+    if type(arr) is not list:
+        raise TypeError('Parameter must of be of type <list>')
+    if not (len(arr[0]) > 1 and len(arr) > 1):
+        raise ValueError('Function expects 2 x 2 list or greater')
+
+    # Initialize our max product
+    max_p = arr[0][0] * arr[1][0]
```

Comment:
So by putting this in you wont have to set your max product to zero but it only allows you to run against certain cases, its not inclusive of all cases.

Reply:
Are you sure about that? Pretty sure that everything with min dimensionality 2x2 is tested, and anything with less dimensionality is not 2D. Also anything less than 2x2 can't safely have a product initialized to 0.